torch/tensor.py

```diff
@@ -280,14 +280,13 @@ def repeat(self, *sizes):
             xsize = [1] + xsize
 
         size = torch.Size([a * b for a, b in zip(xsize, repeats)])
-        xtensor.resize_(torch.Size(xsize))
+        xtensor = xtensor.view(torch.Size(xsize))
```

Comment:
I think this will break if the input tensor is non-contiguous.

Reply:
If the only change this line does is unsqueezeing some dimensions then let’s use unsqueeze instead of view (it works will all kinds of tensors)